server/src/main/java/org/jboss/as/server/operations/HttpManagementAddHandler.java

```diff
@@ -185,6 +183,10 @@ protected void installServices(OperationContext context, HttpInterfaceCommonPoli
             RemotingHttpUpgradeService.installServices(serviceTarget, ManagementRemotingServices.HTTP_CONNECTOR, httpConnectorName,
                     ManagementRemotingServices.MANAGEMENT_ENDPOINT, commonPolicy.getConnectorOptions());
         }
+        if (socketBindingServiceName != null) {
+            return Arrays.asList(socketBindingServiceName, UndertowHttpManagementService.SERVICE_NAME);
+        }
+        return Arrays.asList(ManagementRemotingServices.MANAGEMENT_ENDPOINT, UndertowHttpManagementService.SERVICE_NAME);
```

Comment:
I don't believe anything related to ManagementRemotingServices is required unless http-upgrade is turned on.

On both this and the host controller variant I believe the service that best reflects the requirement for HTTP Upgrade isn't the endpoint service. Rather it's the one installed by RemotingHttpUpgradeService.installServices, named "UPGRADE_SERVICE_NAME.append(remotingConnectorName)". That service depends on the endpoint.